src/main/java/org/broadinstitute/hellbender/utils/read/CigarConversionUtils.java

```diff
@@ -0,0 +1,166 @@
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+
+import java.util.*;
+
+/**
+ * Utilities for bidirectional conversion between SAM Cigar and GA4GH-style List&lt;CigarUnit&gt;.
+ *
+ * Can convert either entire Cigars or individual Cigar elements.
+ *
+ * Conversion from List&lt;CigarUnit&gt; to SAM Cigar can be lossy, since there is no equivalent to
+ * CigarUnit's referenceSequence field in SAM Cigars. This field is silently dropped during conversion,
+ * if present.
+ */
+public final class CigarConversionUtils {
+
+    /**
+     * Valid cigar operators for CigarUnits as defined in the GA4GH spec. The order of these operators matters,
+     * and must match the ordering in {@link #SAM_CIGAR_ELEMENT_OPERATORS} below.
+     */
+    private static final List<String> CIGAR_UNIT_OPERATORS =
+            Arrays.asList("ALIGNMENT_MATCH",
+                          "INSERT",
+                          "DELETE",
+                          "SKIP",
+                          "CLIP_SOFT",
+                          "CLIP_HARD",
+                          "PAD",
+                          "SEQUENCE_MATCH",
+                          "SEQUENCE_MISMATCH");
+
+    /**
+     * Valid cigar operators for SAM Cigars as defined in the SAM spec. The order of these operators matters,
+     * and must match the ordering in {@link #CIGAR_UNIT_OPERATORS} above.
+     */
+    private static final List<CigarOperator> SAM_CIGAR_ELEMENT_OPERATORS =
+            Arrays.asList(CigarOperator.M,
+                          CigarOperator.I,
+                          CigarOperator.D,
+                          CigarOperator.N,
+                          CigarOperator.S,
+                          CigarOperator.H,
+                          CigarOperator.P,
+                          CigarOperator.EQ,
+                          CigarOperator.X);
+
+    /**
+     * Lookup table from CigarUnit operator -> equivalent SAM CigarOperator
+     */
+    private static final Map<String, CigarOperator> cigarUnitOperatorToCigarElementOperatorTable;
+
+    /**
+     * Lookup table from SAM CigarOperator -> equivalent CigarUnit operator
+     */
+    private static final Map<CigarOperator, String> cigarElementOperatorToCigarUnitOperatorTable;
+
+    static {
+        // Populate cigar operator lookup tables
+
+        cigarUnitOperatorToCigarElementOperatorTable = new HashMap<>(CIGAR_UNIT_OPERATORS.size() * 2);
+        cigarElementOperatorToCigarUnitOperatorTable = new HashMap<>(CIGAR_UNIT_OPERATORS.size() * 2);
+
+        for ( int i = 0; i < CIGAR_UNIT_OPERATORS.size(); ++i ) {
+            cigarUnitOperatorToCigarElementOperatorTable.put(CIGAR_UNIT_OPERATORS.get(i), SAM_CIGAR_ELEMENT_OPERATORS.get(i));
+            cigarElementOperatorToCigarUnitOperatorTable.put(SAM_CIGAR_ELEMENT_OPERATORS.get(i), CIGAR_UNIT_OPERATORS.get(i));
+        }
```

Comment:
may want to make the maps unmodifiable at this point at the end of the initializer